### ISSUE TITLE
Dotted outline focus indicators in SC2.4.11 - Minimum area bullet

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag#1369]
Since dotted outline focus indicators are so prevalent on websites, even among sites that already use their own custom indicator rather than th browser default, could we add a brief paragraph to the Understanding doc for this SC to cover them?

It needs to point out that, when a dotted outline is used, there are only half the number of "the pixels that change" in the focused state compared to a solid outline indicator. So in order to still comply with the definition of minimum area and focus indication area ("the pixels that change between the focused and unfocused states"), the thickness of a dotted indicator needs to be at least two pixels thick to ensure the minimum area required by the SC is satisfied.

**Comment by guyhickling** [comment]
NB: Also, as currently noted in the Understanding doc in the caption under Figure 2, there is a slight complexity when the outline (solid or otherwise) is inside the boundary of the control. Figure 2 shows a button with a solid outline of 2px to compensate for the slightly shorter length when inside the control. However that can present problems for very small controls like radio buttons and checkboxes if the outline is inside (some designers do that). And a dotted one would have to be 3px which poses the same difficulty for such small controls. The Understanding doc hasn't addressed this difficulty. 

Perhaps we could address it, in the Understanding, by pointing out that outlines (whether solid or otherwise) are better outside of radio buttons and other very small controls to improve visibility.

**Comment by LiLoDavis** [comment]
> It needs to point out that, when a dotted outline is used, there are only half the number of "the pixels that change" in the focused state compared to a solid outline indicator.

In fact, if the thickness is 4 or greater, the dots become round, which means less than half the pixels change on focus.

**Comment by alastc** [comment]
Hi Guy, I made a proposal in [1348](url), would that do it?

**Comment by guyhickling** [comment]
The part where you say in #1348 "a dotted line will have roughly half the number of pixels due to gaps." is good. But you don't mention anything about 1px lines, which is the critical issue here. You start off with a 2px example which already satisfies the SC. In real life most developers tend to use 1px outlines by default for their focus outline, unless they are deliberately thinking about people with poor eyesight (we are talking about custom outlines here, of course, not leaving it to the browsers' defaults). 

(Also mentioning anti-aliasing just confuses the issue, because dotted outlines are not normally anti-aliased in most browsers - there's no need for it in a straight line. They consist of just a line with one pixel showing the "ink" colour and the next pixel left empty or transparent, and so on alternately.) 

So I would like us to say something to explicitly highlight that a 1px dotted line doesn't meet the minimum area definition. How about (in the Understanding doc):

> When a dotted outline is used, the gaps mean only about half the number of the pixels change color in the focused state compared to a solid outline indicator. So in order to still comply with the definition of minimum area and focus indication area ("the pixels that change between the focused and unfocused states"), a dotted indicator needs to be at least two pixels thick to ensure the minimum area required by the SC is satisfied.

**Comment by alastc** [comment]
> But you don't mention anything about 1px lines, which is the critical issue here. You start off with a 2px example which already satisfies the SC

Of course, if you have a dotted outline then starting with 1px is an automatic fail of the 1st bullet. Any kind of 1px dotted or dashed border cannot meet the size requirement.

It only becomes a question at 2px and above, where it _might_ fulfill it.

**Comment by alastc** [comment]
I'm going to assume this is closed by #1493 (if that is agreed) unless someone says otherwise.

**Comment by guyhickling** [comment]
> Of course, if you have a dotted outline then starting with 1px is an automatic fail of the 1st bullet. 

So why can't we just say that in the Understanding doc? It just needs a single sentence, but it will save a lot of arguing with designers who want to push back on it. Designers whose preferred focus indicator has always been a faint dotted outline. This may be clear to you, but it won't be clear to lots of people out there simply because the Understanding doesn't mention this standard way of doing it that everybody is so used to following.

**Comment by alastc** [comment]
Ok, I've added this to the typical examples:

- A 1 CSS pixel dotted outline around the whole component would **not pass** the size requirement, as it is roughly 50% of the surface area of a solid line.

**Comment by alastc** [comment]
Closing based on #1493

**Comment by guyhickling** [comment]
That's excellent. Many thanks.
